### PR TITLE
Merge hf2 into dev

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

Merge hf2 into dev

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: models/src/main/protobuf/CasperMessage.proto [hunk 1]
```diff
@@ -121,6 +121,7 @@ message BlockMetadataInternal {
   bool invalid                          = 8; // whether the block was marked as invalid
   bool directlyFinalized                = 9; // whether the block has been last finalized block (LFB)
   bool finalized                        = 10;// whether the block is finalized
+  bytes postStateHash                    = 11;// whether the block is finalized
```
Comment (zsluedem) [inline models/src/main/protobuf/CasperMessage.proto]: ~~Something I am not sure~~ It is just metaData

File: models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala [hunk 2]
```diff
@@ -259,7 +259,7 @@ private[sorter] object ExprSortMatcher extends Sortable[Expr] {
       case gs: GString => ScoredTerm(e, Node(Score.STRING, Leaf(gs.value))).pure[F]
       case gu: GUri    => ScoredTerm(e, Node(Score.URI, Leaf(gu.value))).pure[F]
       case GByteArray(ba) =>
-        ScoredTerm(e, Node(Score.EBYTEARR, Leaf(ba.toStringUtf8))).pure[F]
```
Comment (zsluedem) [inline models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala:262]: This is not backwards-compatible

Comment (tgrospic) [reply models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala:262]: That's fine, we are including other breaking changes like channels-map removal. We only need to ensure that we can start test net with single-parent until BM part is finished.

**Comment by tgrospic** [comment]
:bulb: We can keep PRs like this constantly open for feature branches. They are like live update for _feature_ changes with CI running on change to target and source branch.

**Pull request closed without merging**